modules/standard/FileSystem.chpl

```diff
@@ -593,7 +593,9 @@ proc locale.cwd(): string throws {
     var tmp:c_string;
     // c_strings can't cross on statements.
     err = chpl_fs_cwd(tmp);
-    ret = createStringWithOwnedBuffer(tmp);
+    try! {
```

Comment:
more try! in a throwing function

Reply:
My reasoning is not much different than what I had for c_string _cast comment I had above.

Even though this particular instance will change, it'll probably change to something that doesn't throw any error (escaping non-UTF-8 bytes etc).

From a more philosophical standpoint, what is your concern with having `try!` in a `throws` function? The user will assume that the function will never halt and whatever error occurs in it will be either handled or propagated?

Reply:
> From a more philosophical standpoint, what is your concern with having try! in a throws function? The user will assume that the function will never halt and whatever error occurs in it will be either handled or propagated?

yes, generally speaking we should be avoiding halts, and if the function `throws` it is obvious how not to halt.